docs/specification/draft/client/elicitation.mdx

```diff
@@ -151,139 +257,236 @@ To request information from a user, servers send an `elicitation/create` request
+
+<Note>
+  **Important**: Out-of-band elicitation is *not* for re-authorizing the MCP
+  client's access to the MCP server (that is covered by [MCP
+  authorization](../basic/authorization)). It's specifically for scenarios where
+  the MCP server needs to obtain sensitive information, or authorization for
+  third-party services on behalf of the user. The MCP client's authorization
+  (bearer) token remains unchanged throughout this process.
+</Note>
+
+Out-of-band elicitation requests **MUST** specify `mode: "oob"` and include these parameters:
+
+| Name            | Type   | Description                               |
+| --------------- | ------ | ----------------------------------------- |
+| `url`           | string | The URL that the user should navigate to. |
+| `elicitationId` | string | A unique identifier for the elicitation.  |
+
+The `url` parameter **MUST** contain a valid URL. The `message` parameter **MUST NOT** contain a URL.
+
+#### Example: Request Sensitive Data
+
+This example shows an out-of-band elicitation request directing the user to a secure URL where they can provide sensitive information (an API key, for example).
+The same request could direct the user into an OAuth authorization flow, or a payment flow. The only difference is the URL and the message.
+
+**Request:**
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 3,
+  "method": "elicitation/create",
+  "params": {
+    "mode": "oob",
+    "elicitationId": "550e8400-e29b-41d4-a716-446655440000",
+    "url": "https://mcp.example.com/ui/set_api_key",
+    "message": "Please provide your API key to continue."
+  }
+}
+```
+
+**Response:**
 
 ```json
 {
   "jsonrpc": "2.0",
-  "id": 2,
+  "id": 3,
   "result": {
-    "action": "reject"
+    "action": "accept"
+  }
+}
+```
+
+The response with `action: "accept"` indicates that the user has consented to the
+interaction. It does not mean that the interaction is complete. The interaction occurs out
+of band and the client is not aware of the outcome, unless the client requests progress updates.
+
+### Progress Tracking
+
+The client **MAY** request progress updates from the server by sending an `elicitation/track` request with a [progress token](/specification/draft/basic/utilities/progress#progress-token). This is particularly useful in out-of-band mode, because the client is not involved in the interaction.
+
+The client **MUST** include an `elicitationId` in the request to identify which elicitation to send progress updates for. The client **SHOULD** include a `progressToken` in the request's `_meta` field.
+
+The server **MUST** ignore any `elicitation/track` requests containing an `elicitationId` that is not known or does not belong to the client.
+
+The server **MAY** send a `notifications/progress` notification to the client with the progress token and the progress status.
+
+The server **SHOULD** send a `elicitation/track` response to the client once the elicitation is complete.
```

Comment:
Leveraging the progress tracking here adds some complexity to for servers looking to be strictly 'correct'. The server author could rightfully notice that the `elicitation/track` message could arrive for an already-completed elicitation. So the server author has to decide:

1. Do I try to close this little race window by saving EVERY elicitation outcome?
2. How long do I save these outcomes?
3. When is a progress request too old?

Maybe we should be giving some guidance on these topics. I've been trying to think if there's an ergonomic way to totally close any race window (ie: asking for subscription in `accept` response) but what I'm coming up with only shrinks the window without eliminating it.

Reply:
Totally agree that this could be more ergonomic. We were looking at doing something like [resource subscriptions](https://modelcontextprotocol.io/specification/2025-06-18/server/resources#subscriptions) (though that's built on client requests too. 

In the end, for this PR, we decided to leverage the existing progress capability. Assuming this PR merges, I'd like to find a way to make this more ergonomic with you @ggoodman , if you're interested. I think there could be other places in MCP to benefit. 

We purposefully made progress optional here so that any future mechanisms could be adopted alongside the progress approach.

Reply:
👍🏼 don't want to block this landing on this subject. I think it's really nice to see this latest take on the proposal have a "transactional" story built in. We can refine the ergonomics in the future.

Happy to collaborate on a follow-up.

Reply:
Thanks for the good feedback @ggoodman. Following up here - in the final draft (the latest commits to this PR), we dropped `elicitation/track` entirely in favor of a simple server->client [notification](https://modelcontextprotocol.io/specification/2025-06-18/basic#notifications).